module/zfs/zfs_vnops.c

```diff
@@ -3010,29 +3009,27 @@ zfs_setattr(struct inode *ip, vattr_t *vap, int flags, cred_t *cr)
 
 	if (mask & ATTR_MTIME) {
 		ZFS_TIME_ENCODE(&vap->va_mtime, mtime);
+		ZTOI(zp)->i_mtime = timespec_trunc(vap->va_mtime,
+				ZTOI(zp)->i_sb->s_time_gran);
```

Comment:
Ok, goot thing vim's "set list" option shows the discrepancies. Fixed in current version but will push once tests are finished so that I get a baseline results